src/gallia/services/uds/ecu.py

```diff
@@ -525,3 +527,90 @@ async def _request(
 
             if response is not None:
                 await self.update_state(request, response)
+
+
+class TesterPresentSender:
+    """Send TesterPresent after periods of inactivity"""
+
+    def __init__(
+        self,
+        delay: float,
+        send_tester_present: Callable[[], Awaitable[bool]],
+        oneshot: bool = False,
+        ignore_activity: bool = False,
+    ):
+        """
+        delay: idle time in seconds (e.g. 0.5)
+        send_tester_present: async callable that sends the keepalive on your channel
+        oneshot:  if True, send once, then wait for next real activity to re-arm;
+                  if False, keep sending every 'delay' while idle.
+        ignore_activity: ignore activity() calls such that keepalives are sent regardless of activity
+        """
+        self._delay = delay
+        self._send = send_tester_present
+        self._oneshot = oneshot
+        self._ignore_activity = ignore_activity
+        self._activity_event = asyncio.Event()
+        self._responsive = asyncio.Event()
+        self._task: asyncio.Task[None] | None = None
+        self._cancelled: bool = False
+
+    async def reset_timer(self) -> None:
+        """Call this whenever you send or receive real traffic to reset the timer"""
+        if self._ignore_activity is True:
+            return
+        self._activity_event.set()
+
+    async def start(self) -> None:
+        self._cancelled = False
+
+        if self._task is not None:
+            logger.warning("TesterPresentSender already running")
```

Comment:
If `.start()` is called multiple times, isn't this a programming error? Maybe we could raise an `AssertionError` instead of logging a warning.